x-pack/legacy/plugins/reporting/export_types/png/server/execute_job/index.ts

```diff
@@ -5,39 +5,34 @@
  */
 
 import * as Rx from 'rxjs';
-import { i18n } from '@kbn/i18n';
 import { mergeMap, catchError, map, takeUntil } from 'rxjs/operators';
 import { PLUGIN_ID, PNG_JOB_TYPE } from '../../../../common/constants';
+import { ServerFacade, ExecuteJobFactory, ESQueueWorkerExecuteFn } from '../../../../types';
 import { LevelLogger } from '../../../../server/lib';
-import { generatePngObservableFactory } from '../lib/generate_png';
 import {
   decryptJobHeaders,
   omitBlacklistedHeaders,
   getConditionalHeaders,
   getFullUrls,
 } from '../../../common/execute_job/';
+import { JobDocPayloadPNG } from '../../types';
+import { generatePngObservableFactory } from '../lib/generate_png';
 
-export function executeJobFactory(server) {
+export const executeJobFactory: ExecuteJobFactory<
+  JobDocPayloadPNG,
+  ESQueueWorkerExecuteFn<JobDocPayloadPNG>
+> = function executeJobFactoryFn(server: ServerFacade) {
   const generatePngObservable = generatePngObservableFactory(server);
   const logger = LevelLogger.createForServer(server, [PLUGIN_ID, PNG_JOB_TYPE, 'execute']);
 
-  return function executeJob(jobId, jobToExecute, cancellationToken) {
+  return function executeJob(
+    jobId: string,
+    jobToExecute: JobDocPayloadPNG,
+    cancellationToken: any
+  ) {
     const jobLogger = logger.clone([jobId]);
-    const process$ = Rx.of({ job: jobToExecute, server }).pipe(
+    const process$ = Rx.of({ job: jobToExecute, server, logger }).pipe(
       mergeMap(decryptJobHeaders),
```

Comment:
Wow, this is a really nice cleanup

Reply:
It's getting better, but I think there could be focused cleanup of the observable pipelines. 

Related: https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/reporting/export_types/common/lib/screenshots/index.ts#L51

The pipeline cleanup stuff isn't related to NP Migration, but would be nice to handle at some point.